presto-hive/src/main/java/com/facebook/presto/hive/HudiDirectoryLister.java

```diff
@@ -55,11 +56,19 @@ public HudiDirectoryLister(Configuration conf, ConnectorSession session, Table t
     {
         log.info("Using Hudi Directory Lister.");
         this.metadataEnabled = isHudiMetadataEnabled(session);
+        Configuration actualConfig = ((CachingJobConf) conf).getConfig();
+        /*
+        WrapperJobConf acts as a wrapper on top of the actual Configuration object. If `hive.copy-on-first-write-configuration-enabled`
+        is set to true, the wrapped object is instance of CopyOnFirstWriteConfiguration.
+         */
+        if (actualConfig instanceof CopyOnFirstWriteConfiguration) {
+            actualConfig = ((CopyOnFirstWriteConfiguration) actualConfig).getConfig();
```

Comment:
Since HudiDirectoryLister is singleton for the lifecycle of query, so it should be fine doing this once. Can you please confirm through logs or otherwise that it only happens once per query?

Reply:
confirmed, this happens only once per query